### PR TITLE
new-api: implement Pause() and Resume()

### PR DESCRIPTION
**Pull request by avagin**

### CODE REVIEW

File: integration/exec_test.go [hunk 1]
```diff
@@ -300,3 +303,78 @@ func TestEnter(t *testing.T) {
...
+		t.Fatal(err)
+	}
+
+	container, err := factory.Create("test", config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer container.Destroy()
+
+	// Execute a first process in the container
```
Comment (mrunalp) [inline integration/exec_test.go]: Take out this comment as it implies there might be a second process starting after?

Comment (avagin) [inline integration/exec_test.go]: Fixed, thanks.

File: namespaces/exec.go [hunk 2]
```diff
@@ -73,7 +71,11 @@ func Exec(args []string, env []string, console string, command *exec.Cmd, contai
 	if err != nil {
 		return terminate(err)
 	}
-	defer cgroupManager.Destroy()
+	defer func() {
+		if err != nil {
```
Comment (mrunalp) [inline namespaces/exec.go]: Shouldn't we always call cgroups destroy irrespective of err state on exiting this function?

Comment (avagin) [inline namespaces/exec.go]: No, we should not. This function returns a child pid.

Comment (mrunalp) [inline namespaces/exec.go]: Ahh, okay. I see what you mean (looked at the overall flow).

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**